### PR TITLE
Include chain in client node name.

### PR DESCRIPTION
**Pull request by afck**
## Motivation

The `Client` is always called `"Client node"`. This makes debugging difficult in tests involving multiple clients.

## Proposal

Include the chain ID(s) in the name.

## Test Plan

No changed logic, but it should make debugging end-to-end tests easier.

## Release Plan

- Nothing to do / These changes follow the usual release cycle.

## Links

- (Found while debugging https://github.com/linera-io/linera-protocol/pull/2538.)
- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

**Review by ma2bd — COMMENTED** [review]
Why don't we name wallets when we create them? (at random by default and otherwise using a user provided name)

**Comment by afck** [comment]
Good idea, maybe that's more useful, and also more readable: In the tests they could simply be "Client 1" and "Client 2".

File: linera-client/src/unit_tests/chain_listener.rs [hunk 1]
```diff
@@ -161,7 +161,7 @@ async fn test_chain_listener() -> anyhow::Result<()> {
             delivery,
             false,
             [chain_id0],
-            "Client node",
+            format!("Client node for {:.8}", chain_id0),
```
Comment (MathieuDutSik) [inline linera-client/src/unit_tests/chain_listener.rs:164]: It is a minor point, but I find it a little odd that the `Client` has a `name` associated with its constructor. Though of course, this is not part of this PR.

**Review by MathieuDutSik — APPROVED** [review]
Better logging is always welcomed.

File: linera-client/src/unit_tests/chain_listener.rs (hunk 1 above)
Comment (afck) [reply linera-client/src/unit_tests/chain_listener.rs:164]: Yes, like @ma2bd said above: Let's name the _wallets_ instead. I created https://github.com/linera-io/linera-protocol/issues/2551 for that.

**Comment by afck** [comment]
Merging this, to make it easier to debug issues on `main`. I created https://github.com/linera-io/linera-protocol/issues/2551 to name the wallets instead; I'll do that as soon as we're done debugging the failing test.
